File: 典型90/Typical90_007.py
import sys
import bisect
import collections
import heapq
import itertools
import math

input = sys.stdin.readline

def main():
    ans = 0
    N = int(input())
    A = list(map(int, input().split()))
    Q = int(input())

    A.sort()
    for _ in range(Q):
        B = int(input())
        temp = bisect.bisect_left(A, B)
        if temp == N:
            ans = abs(A[-1] - B)
        else:
            ans = min(abs(A[temp] - B), abs(A[temp-1]-B))
        print(ans)

if __name__ == '__main__':
    main()

File: 典型90/Typical90_007.rs
use proconio::{input, fastout};
use std::cmp::min;

#[fastout]
fn main() {
  input!{
    n: usize,
    mut a: [isize; n],
    q: usize,
    b: [isize; q]
  }
  a.push(std::isize::MIN);
  a.push(std::isize::MAX);
  a.sort();
  a.dedup();

  for r in b {
    let idx = a.binary_search(&r).unwrap_or_else(|x| x);
    let ans = min((a[idx]-r).abs(), (a[idx-1]-r).abs());
    println!("{}", ans);
  }
}